rpcs3/Emu/Memory/vm.h

```diff
@@ -146,23 +146,18 @@ namespace vm
 	// Allocate segment at specified location, does nothing if exists already
 	std::shared_ptr<block_t> reserve_map(memory_location_t location, u32 addr, u32 area_size, u64 flags = 0x200);
 
-	// Get PS3/PSV virtual memory address from the provided pointer (nullptr always converted to 0)
+	// Get PS3 virtual memory address from the provided pointer (nullptr or pointer from outside is always converted to 0)
+	// Super memory is allowed as well
 	inline vm::addr_t get_addr(const void* real_ptr)
 	{
-		if (!real_ptr)
-		{
-			return vm::addr_t{};
-		}
-
 		const std::ptrdiff_t diff = static_cast<const u8*>(real_ptr) - g_base_addr;
-		const u32 res = static_cast<u32>(diff);
 
-		if (res == diff)
+		if (diff + 0u <= u64{UINT32_MAX} * 2 + 1)
 		{
-			return static_cast<vm::addr_t>(res);
+			return vm::addr_t{static_cast<u32>(diff)};
 		}
 
-		fmt::throw_exception("Not a virtual memory pointer (%p)", real_ptr);
+		return {};
```

Comment:
This exception helped some people before, who were trying to learn the codebase. What's the reason of removing it?

Reply:
Added vm::try_get_addr for with error status returned, vm::get_addr still throws on invalid arg.